### PR TITLE
Extended rate_transformer_node to non-linear summation

### PR DESCRIPTION
**Pull request by janhahne**
This PR extends the `rate_transformer_node` to non-linear summation.

@jakobj and @heplesser  I was not sure where to put this PR. Since 5g changes the corresponding parts of the code significantly, a PR of the 4g code to the main repository would only result in an unnecessary merge conflict for 5g. You might as well close this PR here if it is not reviewed and merged at the time when 5g is ready for merge to master. However, I would like this to be part of NEST 2.16.0.

### CODE REVIEW

**Comment by jakobj** [comment]
i think it makes sense to merge it directly into 5g, and don't see any reason we can not review it here. @janhahne will you invite people to review this? i suspect @ddahmen would be a good candidate?

**Comment by janhahne** [comment]
@jakobj I also think that the changes are quite straightforward. The default behavior is unchanged, the class is just extended by an additional option which is also available for the other base classes. So, if you and @heplesser agree, then we can merge it in my opinion without consulting anyone else :-) Otherwise David or Mario Senden are good candidates for a review. 

Regarding nonlinearity vs. nonlinearties: There is indeed just `nonlinearties_.input()` used for this class, but the other base class `rate_neuron_ipn` has multiple nonlinearties. I would suggest to keep the name for consistency across rate model base classes.

File: models/rate_transformer_node.h [hunk 1]
```diff
@@ -51,8 +51,11 @@ Name: rate_transformer_node - Rate neuron that sums up incoming rates
 
 Description:
 
-The rate transformer node simply sums up all incoming rates and applies
-the nonlinearity specified in the function input of the template class.
+The rate transformer node simply applies the nonlinearity specified in the
+input-function of the template class to all incoming inputs. The boolean
+parameter linear_summation determines whether the input function is applied to
+the summed up incoming connections (= True, default value) or to each input
+individually (= False).
```
Comment (heplesser) [inline models/rate_transformer_node.h]: I would drop the `= ` in the parentheses.

**Comment by heplesser** [comment]
@janhahne All builds fail during the missing `void` return type, see #98.

**Comment by heplesser** [comment]
@MSenden @ddahmen Could one of you review this asap?

**Comment by janhahne** [comment]
@heplesser Fixed the documentation and added the missing `void` in order to make the tests work.

The additional changes to other files are just clang-format issues.

**Comment by jakobj** [comment]
looks like everyone is happy. merging.

**Pull request merged**